### PR TITLE
Adding forward_kwargs to integrated gradients, Transformers example

### PR DESCRIPTION
**Pull request by gipster**
Main changes:
* An extra argument `forward_kwargs` was added to the `explain` method. This allows to handle models that accept keywords arguments in their `call` method (for example [transformers](https://github.com/huggingface/transformers) models).

Minor changes
* Some simplification of the code
* Tests added  

To do
* Transformers example / tutorial

### CODE REVIEW

File: alibi/explainers/integrated_gradients.py [hunk 1]
```diff
@@ -381,6 +388,7 @@ def _gradients_layer(model: Union[tf.keras.models.Model],
                      orig_dummy_input: Union[list, np.ndarray],
                      x: Union[List[tf.Tensor], tf.Tensor],
                      target: Union[None, tf.Tensor],
+                     forward_kwargs: Optional[dict] = None,
                      compute_layer_inputs_gradients: bool = False) -> tf.Tensor:
```
Comment (jklaise) [inline alibi/explainers/integrated_gradients.py:392]: If we're renaming `compute_layer_inputs_gradients` to `attribute_to_layer_inputs` we should do it everywhere in the module, including the private methods.

File: alibi/explainers/integrated_gradients.py [hunk 2]
```diff
@@ -77,25 +68,8 @@ def _sum_rows(inp):
...
-                                          orig_call,
-                                          orig_dummy_input,
-                                          end_point,
-                                          target,
-                                          run_from_layer_inputs=layer_inputs_attributions)
-    else:
-        start_out = _run_forward(model, start_point, target)
-        end_out = _run_forward(model, end_point, target)
+    start_out = _run_forward(model, start_point, target, forward_kwargs=forward_kwargs)
+    end_out = _run_forward(model, end_point, target, forward_kwargs=forward_kwargs)
```
Comment (jklaise) [inline alibi/explainers/integrated_gradients.py:72]: Can you summarize why the old approach is note needed anymore?

File: examples/integrated_gradients_transformers.ipynb [hunk 3]
```diff
@@ -0,0 +1,1235 @@
+{
```
Comment (jklaise) [inline examples/integrated_gradients_transformers.ipynb:1]: <p>Don't forget to remove before final version.</p><p> </p><p>Also, create a title for the notebook and an intro paragraph what this notebook is about.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/SeldonIO/alibi/pull/430/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='0'/>

Comment (jklaise) [inline examples/integrated_gradients_transformers.ipynb:1]: <p>Add a paragraph explaining briefly the utility functions defined below.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/SeldonIO/alibi/pull/430/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='1'/>

Comment (jklaise) [inline examples/integrated_gradients_transformers.ipynb:1]: <p>session --&gt; section</p><p>do not --&gt; does not</p><p>finetuned --&gt; fine-tuned</p><p>end sentences with full stop please.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/SeldonIO/alibi/pull/430/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='8'/>

Comment (jklaise) [inline examples/integrated_gradients_transformers.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #4.            super(AutoModelWrapper, self).__init__()
</pre><p>parameters are unnecessary, just use <code>super().__init__()</code></p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/SeldonIO/alibi/pull/430/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='11'/>

Comment (jklaise) [inline examples/integrated_gradients_transformers.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #10.            #return out
</pre><p>delete</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/SeldonIO/alibi/pull/430/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='11'/>

Comment (jklaise) [inline examples/integrated_gradients_transformers.ipynb:1]: <p>ingrated --&gt; integrated</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/SeldonIO/alibi/pull/430/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='13'/>

Comment (jklaise) [inline examples/integrated_gradients_transformers.ipynb:1]: <p>End of sentence punctuation.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/SeldonIO/alibi/pull/430/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='14'/>

Comment (jklaise) [inline examples/integrated_gradients_transformers.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #1.    #x_test_sample = ["I love you, I like you",
</pre><p>remove lines</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/SeldonIO/alibi/pull/430/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='15'/>

Comment (jklaise) [inline examples/integrated_gradients_transformers.ipynb:1]: <p>grammar</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/SeldonIO/alibi/pull/430/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='17'/>

Comment (jklaise) [inline examples/integrated_gradients_transformers.ipynb:1]: <p>session --&gt; section</p><p>please rewrite with spelling and grammar accounted for</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/SeldonIO/alibi/pull/430/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='31'/>

Comment (jklaise) [inline examples/integrated_gradients_transformers.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #22.                #self.dropoutl_1 = tf.keras.layers.Dropout(dropout_1)
</pre><p>remove unnecessary lines</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/SeldonIO/alibi/pull/430/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='45'/>

Comment (jklaise) [inline examples/integrated_gradients_transformers.ipynb:1]: <p>We will need a better model for a good example...</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/SeldonIO/alibi/pull/430/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='48'/>

Comment (jklaise) [inline examples/integrated_gradients_transformers.ipynb:1]: <p>grammar</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/SeldonIO/alibi/pull/430/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='49'/>

Comment (jklaise) [inline examples/integrated_gradients_transformers.ipynb:1]: <p>spelling</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/SeldonIO/alibi/pull/430/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='52'/>

File: alibi/explainers/integrated_gradients.py (hunk 2 above)
Comment (gipster) [reply alibi/explainers/integrated_gradients.py:72]: The old approach was more cumbersome, it was only needed because the deltas were calculated inside the `_compute_attribution` methods.

File: examples/integrated_gradients_transformers.ipynb (hunk 3 above)
Comment (gipster) [reply examples/integrated_gradients_transformers.ipynb:1]: The notebook is only a stab with minimal quick comments, I know it must be rewritten and check.  But thanks for the spell checking Janis.

Comment (gipster) [reply examples/integrated_gradients_transformers.ipynb:1]: <p>This is only trained with 150 samples for testing. In the final version will be trained with the whole dataset and reaches about 80% accuracyy.</p>

File: examples/integrated_gradients_transformers.ipynb [hunk 4]
```diff
@@ -0,0 +1,1179 @@
...
```
Comment (jklaise) [inline examples/integrated_gradients_transformers.ipynb:1]: <p>"We train a convolutional neural network..." is not accurate, copied from the other notebook?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/SeldonIO/alibi/pull/430/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='1'/>

Comment (jklaise) [inline examples/integrated_gradients_transformers.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #33.        return [z['input_ids'], z['attention_mask']]
</pre><p>Typo here, double return.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/SeldonIO/alibi/pull/430/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='4'/>

Comment (jklaise) [inline examples/integrated_gradients_transformers.ipynb:1]: <p>od --&gt; of</p><p>mail --&gt; main in 2 places</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/SeldonIO/alibi/pull/430/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='15'/>

Comment (jklaise) [inline examples/integrated_gradients_transformers.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #1.    bl = auto_model.layers[0].layers[0].transformer.layer[1]
</pre><p>Please add a sentence or two explaining which layer we're explaining and how to access/find it using from the transformer model.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/SeldonIO/alibi/pull/430/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='17'/>

Comment (gipster) [reply examples/integrated_gradients_transformers.ipynb:1]: Ya I did. Forgot to take it out.

Comment (gipster) [reply examples/integrated_gradients_transformers.ipynb:1]: <p>I added  'The auto model consists of a main BERT layer (layer 0) followed by two dense layers.&nbsp;</p><p>We extract the first transformer's block in the main BERT layer.'</p><p> I honestly don't know how to explain it better. </p>

File: examples/integrated_gradients_transformers.ipynb [hunk 5]
```diff
@@ -0,0 +1,1061 @@
...
```
Comment (jklaise) [inline examples/integrated_gradients_transformers.ipynb:1]: <p>Typos and grammar:</p><p>performances --&gt; performance</p><p>behind the scope --&gt; beyond the scope</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/SeldonIO/alibi/pull/430/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='29'/>

**Review by jklaise — APPROVED** [review]
